### ISSUE TITLE
复读现象

### ISSUE THREAD

**Issue by LaFeuilleMorte** [Liuziyu77/Visual-RFT#29]
Hi, 我在我的数据集上跑，发现模型出现了复读现象，正常的应该是：
content: <think>Let's analyze the described scenario of the image. The main focus is subway station elements. The given objects seem specific to a subway or train environment with a green exit sign, a clock, a map below it, and an overhead fan. Considering this, there's a human in the image. </think>

<answer> [{'bbox_2d': [608, 446, 636, 476], 'label': 'person'}]</answer>
sol: <answer>[{'Position': [937, 943, 988, 999], 'Confidence': 1}]</answer>
student_answer_bbox: [{'bbox_2d': [608, 446, 636, 476], 'label': 'person'}]
ground_truth_bbox: [{'Position': [937, 943, 988, 999], 'Confidence': 1}]

但是我的却是：

content: Detect all objects belonging to the category 'Mailbox Cluster' in the image, and provide the bounding boxes and confidence (between 0 and 1, with two decimal places).
If no object belonging to the category 'Mailbox Cluster' in the image, return 'No Objects'.
Output the thinking process in <think> </think> and final answer in <answer> </answer> tags.The output answer format should be as follows:
<think> ... </think> <answer>[{'Position': [x1, y1, x2, y2], 'Confidence': number}, ...]</answer>
Please strictly follow the format.
assistant
<think> The image contains a set of mailboxes arranged in a grid pattern labeled with a mailbox cluster system. Each cluster appears to be a set of individual mailboxes with numbers on them, likely for tracking and retrieval of mail. </think>
<answer>[{'Position': [25, 62, 894, 700], 'Confidence': 0.987}, {'Position': [24, 717, 466, 803], 'Confidence': 0.987}]</answer>

sol: <answer>[{'Position': [22.157688101287953, 61.57910536670979, 1163.6158075278545, 637.7560662196794], 'Confidence': 1}]</answer>

student_answer_bbox: None

ground_truth_bbox: [{'Position': [22.157688101287953, 61.57910536670979, 1163.6158075278545, 637.7560662196794], 'Confidence': 1}]

可以看到，模型将我的问题又重复了一遍，输出出来了，Base Model是 Qwen2.5-vl-3B，这会导致答案的提取，以及格式的匹配全部失败。

**Comment by Liuziyu77** [comment]
It seems that Qwen-2.5-VL uses absolute coordinates, while Qwen-2-VL uses normalized coordinates in the [0, 1000] range. The training and testing code needs to be modified accordingly to ensure compatibility.

**Comment by LaFeuilleMorte** [comment]
Update: I already uses the absolute coordinate in my datasets. And I change my image resolution (1024*768 -> 512*384), And the problem disappears:

content: <think> The image shows a residential noticeboard with a map of a housing development. The layout includes housing blocks, roads, and a few public amenities such as a park and a train station. The positions and sizes of the housing blocks and amenities can give insights into the layout's planning and layout quality.</think>
<answer>[{'Position': [680, 300, 760, 360], 'Confidence': 0.5}, {'Position': [500, 380, 600, 440], 'Confidence': 0.8}, {'Position': [700, 400, 760, 460], 'Confidence': 0.7}, {'Position': [820, 420, 880, 480], 'Confidence': 0.9}, {'Position': [920, 440, 1000, 500], 'Confidence': 1.0}]</answer>

sol: <answer>[{'Position': [0, 0, 1000, 1000], 'Confidence': 1}]</answer>

student_answer_bbox: [{'Position': [680, 300, 760, 360], 'Confidence': 0.5}, {'Position': [500, 380, 600, 440], 'Confidence': 0.8}, {'Position': [700, 400, 760, 460], 'Confidence': 0.7}, {'Position': [820, 420, 880, 480], 'Confidence': 0.9}, {'Position': [920, 440, 1000, 500], 'Confidence': 1.0}]

ground_truth_bbox: [{'Position': [0, 0, 1000, 1000], 'Confidence': 1}]

iou_results: [(0, 1.0), (0, 0.9), (0, 0.8), (0, 0.7), (0, 0.5)]


That's so weird!!!

**Comment by dyh** [comment]
分类 也遇到了，我把 max_new_tokens=256 ，然后不让它返回思考，就不重复了：
```
    question = (
        "This is an image containing a flower. Please identify the species of the flower based on the image.\n"
        "Output the answer in <answer> </answer> tags."
        "The output answer format should be as follows:\n"
        "<answer>species name</answer>\n"
        "Please strictly follow the format."
    )
```

推理一张退庵的完整代码：

```
import io
import os
import re
import json
import torch
import numpy as np
from PIL import Image
from tqdm import tqdm

torch.manual_seed(1234)

from transformers import Qwen2VLForConditionalGeneration, AutoTokenizer, AutoProcessor
from qwen_vl_utils import process_vision_info
import torch
from transformers import Qwen2VLForConditionalGeneration, AutoProcessor
import re
import sys

# 设定模型路径
model_path = "./share_models/Qwen2-VL-2B-Instruct_save_output/checkpoint-51"  # 权重路径

def load_model_and_processor():
    """加载模型和处理器"""
    model = Qwen2VLForConditionalGeneration.from_pretrained(
        model_path,
        torch_dtype=torch.bfloat16,
        attn_implementation="flash_attention_2",
        device_map="auto",
    )
    processor = AutoProcessor.from_pretrained(model_path, use_fast=True)
    model.eval()
    return model, processor

def infer_image(model, processor, image_path):
    """对单张图片进行推理"""
    question = (
        "This is an image containing a flower. Please identify the species of the flower based on the image.\n"
        "Output the answer in <answer> </answer> tags."
        "The output answer format should be as follows:\n"
        "<answer>species name</answer>\n"
        "Please strictly follow the format."
    )

    # 准备消息
    messages = [
        {
            "role": "user",
            "content": [
                {"type": "image", "image": image_path}
            ] + [{"type": "text", "text": question}],
        }
    ]

    # 处理输入
    text = processor.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
    image_inputs, _ = process_vision_info(messages)

    inputs = processor(
        text=[text],
        images=image_inputs,
        padding=True,
        return_tensors="pt",
    )

    inputs = inputs.to(model.device)

    # 推理
    generated_ids = model.generate(**inputs, max_new_tokens=256)
    generated_ids_trimmed = [
        out_ids[len(in_ids):] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
    ]
    response = processor.batch_decode(
        generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=False
    )[0]

    return response

def extract_answer(response):
    """从模型响应中提取答案"""
    try:
        match = re.search(r"<answer>(.*?)</answer>", response)
        return match.group(1) if match else "No answer found"
    except Exception as e:
        return str(e)

def main(image_path):
    model, processor = load_model_and_processor()
    response = infer_image(model, processor, image_path)
    answer = extract_answer(response)
    print("Model Response:", response)
    print("Extracted Answer:", answer)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python script.py <path_to_image>")
        sys.exit(1)

    image_path = sys.argv[1]
    main(image_path)

```

**Comment by YuZijian** [comment]
https://github.com/Liuziyu77/Visual-RFT/blob/6ad9a6af9dd7c47c7f4c036a40e4dc88b39e7178/src/virft/src/open_r1/trainer/grpo_trainer.py#L398C9-L409C70
